lms/djangoapps/grades/subsection_grade_factory.py

```diff
@@ -43,14 +43,14 @@ def create(self, subsection, read_only=False):
             if assume_zero_if_absent(self.course_data.course_key):
                 subsection_grade = ZeroSubsectionGrade(subsection, self.course_data)
             else:
-                subsection_grade = SubsectionGrade(subsection).init_from_structure(
-                    self.student, self.course_data.structure, self._submissions_scores, self._csm_scores,
+                subsection_grade = SubsectionGrade.create(
+                    subsection, self.course_data.structure, self._submissions_scores, self._csm_scores,
                 )
                 if should_persist_grades(self.course_data.course_key):
                     if read_only:
                         self._unsaved_subsection_grades[subsection_grade.location] = subsection_grade
                     else:
-                        grade_model = subsection_grade.create_model(self.student)
+                        grade_model = subsection_grade.update_or_create_model(self.student)
```

Comment:
This is really minor, but should a subsection grade have its own notion of the learner whose grade it is? It seems off to construct a subsection grade, then later have to tell the grade whose grade it is.

Reply:
interesting thought.  yes, I do see multiple lines of code in subsection_grade.py that requires the student/user info.  let's keep this in mind as we continue to evolve/refine this "core" code.